charts/opentelemetry-collector/values.yaml

```diff
@@ -110,6 +110,36 @@ presets:
   # Best used with mode = daemonset.
   kubeletMetrics:
     enabled: false
+  # Configures the collector to collect MySQL metrics .
+  mysql:
+    # metrics enables collection of MySQL metrics.
+    metrics:
+      enabled: false
+      # If you have multiple instance you'd like to collect, specify each instance in the list below.
+      # If you have only one instance, you don't need to specify the labelSelectors.
+      instances:
+      - labelSelectors:
+            # app.kubernetes.io/name: "mysql"
+            # app.kubernetes.io/instance: "mysql-logs"
+        username: "root"
+        password: ${env:MYSQL_PASSWORD}
```

Comment:
Agree with @povilasv

Reply:
That is the idea basically, to reference the secret that can be provided from `extraEnvs`, similarly to how we do it in `otel-agent` and Coralogix key (but in this case it will be read directly by collector from env var). This gives user the flexibility to specify it in env or other way (e.g. for quick test). We can then specify this or document in `otel-agent` subchart.

Reply:
@nicolastakashi what would be an alternative to avoid that? AFAIK this pattern is used commonly with the collector.

Reply:
Usually, mount the secret as a volume and read the secret from file.

Reply:
I'm not sure if that's possible with collector components. As we chatted privately, I think this is only viable option for now but we can revisit in the future if this becomes a security problem.